src/openms/source/CONCEPT/QCMBRalignment.cpp

```diff
@@ -11,90 +11,165 @@ QCMBRalignment::~QCMBRalignment()
+        
+     if(numOfcorrRT>0)
+     {
+        MzTabPeptideSectionRows PepROWS;
+        for(int i = 0; i< numOfcorrRT;i++)
+        {
+            MzTabPeptideSectionRow PepROW;
+            MzTabString PepSeq;
+            MzTabDouble corrRT;
+            MzTabDouble oriRT;
+                
+            if (spectra[i]=="0") 
+            {
+              
+              PepSeq.set(sequences[i]);
+              corrRT.set(correctedRTs[i]);
+              //oriRT.set(originalRTs[i]);
+              vector<MzTabDouble> cRTs;
+              cRTs.push_back (corrRT);
+              //cRTs.push_back (oriRT);
+              MzTabDoubleList listC;
+              listC.set(cRTs);
+              PepROW.sequence = PepSeq;
+              PepROW.retention_time = listC;
+                
+                //typedef std::pair<String, MzTabString> MzTabOptionalColumnEntry
+              String ori = to_string(originalRTs[i]);
+              MzTabString str = MzTabString(ori);
+              MzTabOptionalColumnEntry oRT = make_pair("original_retention_time",str);
+              vector<MzTabOptionalColumnEntry> v;
+              v.push_back (oRT);
+                
+              MzTabString name = MzTabString(rawFiles[i]);
+              MzTabOptionalColumnEntry sraw = make_pair("raw_source_file",name);
+              v.push_back (sraw);
+              PepROW.opt_ = v;
+                
+              PepROWS.push_back(PepROW);
+
+           }  
+                
+           else if (spectra[i]!="0") 
+           { 
+             //vector<String> rawFiles;
+  	         //vector<String> sequences;
+  	         //vector<float> originalRTs;
+  	         //vector<float> correctedRTs;
+  	         //vector<String> spectra;
+  	  
+             MzTabSpectraRef ref;
+             ref.setSpecRef(spectra[i]);
+             String out_ref = ref.getSpecRef();
+             PepSeq.set(sequences[i]);
+             corrRT.set(correctedRTs[i]);
+             vector<MzTabDouble> cRTs;
+             cRTs.push_back (corrRT);
+             MzTabDoubleList listC;
+             listC.set(cRTs);
+             PepROW.sequence = PepSeq;
+             PepROW.retention_time = listC;
+             PepROW.spectra_ref = ref;
+                
+                //typedef std::pair<String, MzTabString> MzTabOptionalColumnEntry
+             String ori = to_string(originalRTs[i]);
+             MzTabString str = MzTabString(ori);
+             MzTabOptionalColumnEntry oRT = make_pair("original_retention_time",str);
+             vector<MzTabOptionalColumnEntry> v;
+             v.push_back (oRT);
+                
+             MzTabString name = MzTabString(rawFiles[i]);
+             MzTabOptionalColumnEntry sraw = make_pair("raw_source_file",name);
+             v.push_back (sraw);
+             PepROW.opt_ = v;
+                
+             PepROWS.push_back(PepROW);
+
+          }  
+                
+
+       }
+    mztab.setPeptideSectionRows(PepROWS);
```

Comment:
you should not overwrite what is already there, but augment it.
Hence its important to agree on a unique way to identify and match the rows already present.